File: C++ Project/.vscode/Stack/Stack.cpp
#include <iostream>
using namespace std;

class Stack{

public:
    int *arr;
    int top;
    int size;

    Stack(int size)
    {
        this->size = size;
        arr = new int[size];
        top = -1;
    }

public:
    int peek() { return arr[top]; }

    void push(int element)
    {
        if (size - top > 1)
            arr[++top] = element;
        else
            cout << "Stack Overflow" << endl;
    }

    int pop()
    {
        if (top >= 0)
            return arr[top--];
        else
            cout << "Stack is Empty" << endl;
        return -1;
    }

    bool isEmpty()
    {
        if (top == -1)
            return true ;
        else 
            return false;
    }
};

int main(int argc, char const *argv[])
{

    Stack demo(5);

    demo.push(5);
    demo.push(3);

    cout << endl << demo.peek() << endl << endl;

    return 0;
}

File: DSA in Java and Kotlin/src/javaLanguage/stackQuestions/Stack.java
package javaLanguage.stackQuestions;


import java.util.ArrayList;

public class Stack<T> {

    ArrayList<T> arr;
    int size = 0;
    int top;

    public Stack() {
        arr = new ArrayList<>();
        top = -1;
    }

    public Stack(int size) {
        this.size = size;
        arr = new ArrayList<>(size);
        top = -1;
    }

    public T peek() {
        if (top >= 0) return arr.get(top);
        return null;
    }

    public void push(T element) {
        if (size - top >= 1) {
            arr.add(element);
            size++; top++;
        } else throw new IllegalStateException("Size overflow");
    }

    public T pop() {
        if (size <= 0) {
            throw new IllegalStateException("Size overflow");
        } else {
            top--;
            return arr.remove(--size);
        }
    }

}


